Directory.Packages.props

```diff
@@ -3,19 +3,18 @@
     <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
   </PropertyGroup>
   <ItemGroup>
-    <PackageVersion Include="coverlet.collector" Version="3.1.2" />
     <PackageVersion Include="FluentAssertions" Version="6.9.0" />
     <PackageVersion Include="GitHubActionsTestLogger" Version="2.0.1" />
-    <PackageVersion Include="Microsoft.Build.Locator" Version="1.5.5" />
+    <PackageVersion Include="Microsoft.Build.Framework" Version="17.3.1" />
     <PackageVersion Include="Microsoft.Build.Utilities.Core" Version="17.3.1" />
     <PackageVersion Include="Microsoft.Build" Version="17.3.1" />
```

Comment:
This isn't the worth thing, since we're not doing anything crazy with MSBuild that I would expect to change across versions. Self-hosting MSBuild should be fine. This is because we're sort of 'shimming' the expected properties that the SDK props and targets provide in our very limited-scope tests that do Project evaluation/Target execution.

As soon as we want to start trying to in-memory load _actual_ SDK projects, though, we will fail because we're not loading the SDK MSBuild. I don't _think_ we will start doing this anytime soon though - we can get good coverage via shim-hosting like we already do, and end-to-end tests can generate and use the package, driving builds via the CLI.